src/kudu/consensus/consensus_queue.cc

```diff
@@ -1405,6 +1405,7 @@ void PeerMessageQueue::BeginWatchForSuccessor(
   std::lock_guard<simple_spinlock> l(queue_lock_);
 
   transfer_context_ = std::move(transfer_context);
+  successor_watch_peer_notified_ = false;
 
   if (successor_uuid && FLAGS_synchronous_transfer_leadership &&
       PeerTransferLeadershipImmediatelyUnlocked(successor_uuid.get())) {
```

Comment:
With this flags being set by default (as an optimization in 5.6), the logic here is not going to add much value.

Reply:
Even with that flag set, won't we still proceed to do async transfers if the instance is not caught up?

The logic for this PR is so that if the plugin eventually times out waiting for a no-op, it can come back to kudu to cancel the transfer while asking if we've contacted anyone else.